GSoC25_H/src/llm_core.py

```
@@ -0,0 +1,141 @@
+        while retries < self.config.max_retries:
+            try:
+                response = self.client.chat(
+                    model=self.config.name,
+                    messages=messages,
+                    options={
+                        "temperature": self.config.temperature,
+                        "top_p": self.config.top_p,
+                        "num_predict": self.config.num_predict
+                    }
+                )
+                return response
```

Ruff (0.14.13)

Consider moving this statement to an `else` block

(TRY300)